### PR TITLE
feat: implement data platform

### PR DESCRIPTION
**Pull request by mimihalescu**
## What does this fix or implement?

Implemented Data Platform product into the Ansible collection. New modules added:

- dataplatform_cluster
- dataplatform_cluster_config
- dataplatform_cluster_info
- dataplatform_nodepool
- dataplatform_nodepool_info

## Checklist

- [x] PR name added as appropriate (e.g. feat/fix/doc/test/refactor/etc)
- [x] Tests added or updated
- [x] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any

### CODE REVIEW

File: plugins/modules/dataplatform_nodepool.py [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+  - name: Update Data Platform nodepool
+    dataplatform_nodepool:
+      name: "{{ name }}"
+      cluster: "ed67d8b3-63c2-4abe-9bf0-073cee7739c9"
+      node_count: 1
+      cores_count: "1"
+      maintenance_window:
+        day_of_the_week: 'Tuesday'
+        time: '13:03:00'
+      auto_scaling:
```
Comment (rmocanu-ionos) [inline plugins/modules/dataplatform_nodepool.py:227]: I think auto_scaling needs to be added to OPTIONS

File: plugins/modules/dataplatform_nodepool.py [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+
+EXAMPLE_PER_STATE = {
+  'present' : '''
+  - name: Create Data Platform nodepool
+    dataplatform_nodepool:
+      name: "{{ name }}"
+      cluster: "a0a65f51-4d3c-438c-9543-39a3d7668af3"
+      node_count: "1"
+      cpu_family: "AMD_OPTERON"
+      cores_count: "1"
```
Comment (rmocanu-ionos) [inline plugins/modules/dataplatform_nodepool.py:211]: this should be an int in the examples

File: plugins/modules/dataplatform_nodepool.py [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+      ram_size: "2048"
+      availability_zone: "AUTO"
+      storage_type: "SSD"
+      storage_size: "100"
```
Comment (rmocanu-ionos) [inline plugins/modules/dataplatform_nodepool.py:215]: this too should be an int

File: plugins/modules/dataplatform_nodepool.py (hunk 2 above)
Comment (rmocanu-ionos) [reply plugins/modules/dataplatform_nodepool.py:211]: same for ram_size, node_count and storage_size

File: plugins/modules/dataplatform_nodepool.py [hunk 4]
```diff
@@ -0,0 +1,582 @@
...
+    module = AnsibleModule(argument_spec=get_module_arguments(), supports_check_mode=True)
+
+    if not HAS_SDK:
+        module.fail_json(msg='ionoscloud_dataplatform is required for this module, run `pip install ionoscloud_dataplatform`')
+
+    state = module.params.get('state')
+    with ApiClient(get_sdk_config(module, ionoscloud_dataplatform)) as api_client:
+        api_client.user_agent = DATAPLATFORM_USER_AGENT
+        check_required_arguments(module, state, OBJECT_NAME)
+        if state == 'present' and not module.params.get('node_count') and not module.params.get('auto_scaling'):
```
Comment (rmocanu-ionos) [inline plugins/modules/dataplatform_nodepool.py:566]: auto_scaling should be removed

File: plugins/modules/dataplatform_nodepool.py (hunk 1 above)
Comment (mimihalescu) [reply plugins/modules/dataplatform_nodepool.py]: Done ✔️  Thank you!

File: plugins/modules/dataplatform_nodepool.py (hunk 2 above)
Comment (mimihalescu) [reply plugins/modules/dataplatform_nodepool.py]: Done ✔️  Thank you!

File: plugins/modules/dataplatform_nodepool.py (hunk 4 above)
Comment (mimihalescu) [reply plugins/modules/dataplatform_nodepool.py]: Done ✔️  Thank you!

**Review by rmocanu-ionos — APPROVED** [review]

**Pull request merged**